src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Networking/INetworkConnection.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+
+namespace Microsoft.MixedReality.SpectatorView
+{
+    /// <summary>
+    /// Classes that implement this interface represent network connections and allow sending messages across a network.
+    /// </summary>
+    public interface INetworkConnection
+    {
+        /// <summary>
+        /// IP address for the connected peer
+        /// </summary>
+        string Address { get; }
+
+        /// <summary>
+        /// Returns true if the network connection is connected, otherwise false
+        /// </summary>
+        bool IsConnected { get; }
+
+        /// <summary>
+        /// Call to set the network connection state to disconnected
+        /// </summary>
+        void Disconnect();
+
+        /// <summary>
+        /// Checks whether the associated client is still active. If not, the client is disconnected.
+        /// </summary>
+        /// <param name="currentTime">Time to use relative to last active timestamp to determine whether to disconnect</param>
+        void CheckConnectionTimeout(DateTime currentTime);
+
+        /// <summary>
+        /// Call to start enqueuing incoming messages
+        /// </summary>
+        /// <param name="incomingQueue">Queue used for enqueuing messages</param>
+        void QueueIncomingMessages(ConcurrentQueue<IncomingMessage> incomingQueue);
```

Comment:
this method reads in a bit of an unclear manner, but basically it sets up a queue so that you can start receiving messages from the connection. It's a bit unclear what happens if multiple queues are provided. For now, I think we should do a TrySubscribeToIncomingMessages and return false if a subscription already exists. Only our TCPConnectionManager should be providing a queue currently. overtime we can change this as needed but having each component subscribe to viewing this data seems like bad practice.

Reply:
May I suggest something like SetMessageQueue() then?